### PR TITLE
`run()`  instead of `integrate()` and `tabulate()`

### PR DESCRIPTION
**Pull request by stepan-tsirkin**
The idea that was raised in #125, developed further.

the new function `run()` substitutes the old `integrate()` and `tabulate()`.  So, it should be possible to integrate and tabulate in one shot.  The idea that every quantity is a `calculator` that is an object that can be called with a single parameter `data_K` and returns a `Result` of some type. 
the `Result` knows by itself how to save it to file, so the `run`

TODO:

- [ ] rewrite all functionalities as `calculators`
- [x] make it work for tabulation
- [ ] design a format for saveing tabukation result as `npz`
- [ ] make sure things work when different Efermi is used for different quantities (or add a check that disables that)

Open Questions:

1. What to do with tabulation under iterations 

    a )  stop after zeros iteration
    b )  save at every iteration npz files containing added points
    c )  give an option to switch between two modes above

2. what to do with smoothing?
    a) add it to definition of calculators?
    b) other

Basically, after implementing tis, the following files will become useless:

- `__main.py`
- `__integrate.py`
- `__tabulate.py`
- `__evaluate.py`

However, we will keep it for some more time for the fans of the old interface. But maybe will remove in the version 1.0.0

### CODE REVIEW

**Comment by stepan-tsirkin** [comment]
@jaemolihm , do you have any global suggestions on how to develop this PR ? except the usual issues  - to document and to test everything properly ?

**Comment by jaemolihm** [comment]
I skimmed the code, and I think the implementation is good.

Only a single comment: if I am correct, there is a hierarchy of imports like `fermiocean_dynamic -> calculator -> fermiocean`, while I expect `calculator -> {fermiocean_dynamic, fermiocean}`, so that users (who are not developers) only need to deal with `calculator`.

**Comment by jaemolihm** [comment]
I agree with your plan.
